exercises/error-handling/src/test/java/ErrorHandlingTest.java

```diff
@@ -0,0 +1,152 @@
+import junit.framework.AssertionFailedError;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public class ErrorHandlingTest {
+
+    private ErrorHandling errorHandling;
+    private Set<Class> checkedExceptions;
+    private Set<Class> uncheckedExceptions;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        errorHandling = new ErrorHandling();
+        setUpCheckedExceptions();
+        setUpUncheckedExceptions();
+    }
+
+    @Test
+    public void testThrowException() throws Exception {
+        thrown.expect(Exception.class);
+        errorHandling.handleErrorByThrowingException();
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testThrowExceptionWithDetailMessage() throws Exception {
+        thrown.expect(Exception.class);
+        thrown.expectMessage("This is the detail message.");
+        errorHandling.handleErrorByThrowingExceptionWithDetailMessage("This is the detail message.");
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testThrowIllegalArgumentException() throws Exception {
+        thrown.expect(IllegalArgumentException.class);
+        errorHandling.handleErrorByThrowingIllegalArgumentException();
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testThrowIllegalArgumentExceptionWithDetailMessage() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("This is the detail message.");
+        errorHandling.handleErrorByThrowingIllegalArgumentExceptionWithDetailMessage("This is the detail message.");
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testThrowCustomCheckedException() {
+        try {
+            errorHandling.handleErrorByThrowingCustomCheckedException();
```

Comment:
You should be able to deal with this in the same way as the other tests, i.e.
```
thrown.expect(CustomCheckedException.class);
```
The same goes for the next three tests🙂

Reply:
The purpose of the first test is to test the users capacity for throwing an Exception. The user will have to become aware about the need of adding ```throws Exception``` to the method signature. There is no way to bypass the declaration of ```throws Exception``` for the user as long as the test is expecting an instance of Exception.class. I could remove this test and replace it with other two ```testThrowCheckedException``` and ```testThrowUncheckedException```, what do your think? 🙂

I think is a good idea to add these hints 🙂

Reply:
Yes, my problem is with having the test expect the user to throw just `Exception` and not a more specific class. Replacing this test with the other two sounds like a great idea 🙂 Though maybe the `IllegalArgumentException` tests should be before the custom exception tests? That way the user can first use an exception class which already exists before having to worry about custom exceptions. I don't think it matters too much though, either is fine 🙂